controllers/metallb_controller.go

```diff
@@ -67,12 +68,30 @@ func (r *MetallbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
+	err = status.Update(context.TODO(), r.Client, instance, status.ConditionProgressing, "", "")
```

Comment:
I believe this will cause continuous reconciliation.
We need to check the current state (of the deployment) AND the current status and update only if they differ, otherwise what will happen here is a bunch of updates on the cr will trigger reconciliations again and again

Reply:
Modified. We only update the status when it's different. We also check if the reconciliation will do any changes, and only modify the status if it does.